client/src/app/panel/__tests__/PanelSpec.js

```diff
@@ -77,6 +77,47 @@ describe('<Panel>', function() {
       expect(wrapper.find('.foo')).to.have.length(1);
     });
 
+
+    it('should update Tab on prop changes', async function() {
+
+      // given
+      class CustomComponent extends Component {
+        constructor(props) {
+          super(props);
+          this.state = { label: 'Foo' };
+        }
+
+        render() {
+          const { label } = this.state;
+
+          return <Fill slot="bottom-panel" label={ label } id="foo">
+            <div className={ 'foo' } />
+          </Fill>;
+        }
+      }
+
+      const wrapper = renderPanel({
+        children: <CustomComponent />
+      });
+
+      // assume
+      expect(wrapper.find('.panel__link')).to.have.length(1);
+      expect(wrapper.find('.panel__link--active')).to.have.length(1);
+
+      expect(wrapper.find('.panel__link').at(0).find('.panel__link-label').text()).to.equal('Foo');
+      expect(wrapper.find('.panel__link').at(0).hasClass('panel__link--active')).to.be.true;
+
+      // when
+      wrapper.find(CustomComponent).setState({ label: 'Bar' });
+
+      // then
+      return expectEventually(() => {
+        expect(wrapper.find('.panel__link').at(0).find('.panel__link-label').text()).to.equal('Bar');
+        expect(wrapper.find('.panel__link').at(0).hasClass('panel__link--active')).to.be.true;
```

Comment:
This one is redundant, right?

Reply:
We want to check that the Label has changed but is still selected and active